ui/pages/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -303,6 +303,7 @@ export default class ConfirmTransactionBase extends Component {
 
     const notMainnetOrTest = !(isMainnet || process.env.IN_TEST);
     const gasPriceFetchFailure = isEthGasPrice || noGasPrice;
+    const requestingOrigin = new URL(txData.origin)?.hostname;
```

Comment:
`txData` won't always have an origin. I think this is why e2e tests are failing

Reply:
Good point, I've made this more fault tolerant 👍